### PR TITLE
More learning

### PR DESCRIPTION
**Pull request by jml**
Also add a README

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+=============================
+Haskell SPAKE2 implementation
+=============================
+
+Implementation of SPAKE2 key exchange protocol.
+
+Status
+======
+
+Doesn't actually work. We are in the middle of implementing it.
```
Comment (exarkun) [inline README.rst]: Very considerate.

How do you feel about semantic linefeed?

Comment (jml) [inline README.rst]: I don't know what that means.

Comment (jml) [inline README.rst]: I looked this up. Seems like a great idea, but I've never tried it before so I'll probably get it wrong. Happy to try it though.

**Comment by exarkun** [comment]
Linefeed changes look good.

**Pull request merged**